File: chrome-extension/src/services/slash.js
import { authService } from '@/services/auth.js'
import { CONFIG } from '@/config/index.js'

/**
 * 读取 Slash 目录（skills + mcp 占位）的服务模块。
 *
 * 说明：
 * - 此模块只负责“slash 非流式目录接口”，与 `nexus.js` 的 agent 流式能力解耦。
 * - 前端调用 API 模块地址，再由 API 代理到 Nexus，避免浏览器侧跨服务直连。
 */

/**
 * 请求 API 模块的 slash 目录接口并返回目录数据。
 *
 * 调用链路：
 * chrome-extension -> API(`/fastflow/api/v1/clash/catalog`) -> Nexus(`/fastflow/nexus/v1/slash/catalog`)
 *
 * @returns {Promise<{skills: Array, mcp: Array}>}
 */
async function getSlashCatalog() {
  // Authorization 由登录模块统一维护，这里只做透传。
  const token = await authService.getToken() || ''
  const response = await fetch(`${CONFIG.API_BASE_URL}/fastflow/api/v1/clash/catalog`, {
    method: 'GET',
    headers: { 'Authorization': token }
  })

  // 只解析一次响应体，避免 “body stream already read” 错误。
  const payload = await response.json().catch(() => null)
  if (!response.ok) {
    throw new Error(payload?.message || `Request failed with status ${response.status}`)
  }
  if (!payload || payload.code !== 200) {
    throw new Error(payload?.message || 'get slash catalog failed')
  }

  // 统一返回目录结构，调用方只关心 data 层业务字段。
  return payload.data || { skills: [], mcp: [] }
}

export const Slash = {
  getSlashCatalog
}

File: nexus/api/slash.py
"""
Slash 目录接口。

用途：
1) 提供 `/` 上拉面板所需的 skills 列表；
2) 返回 MCP 占位分组，便于前端提前渲染分组结构；
3) 除 agent/builder 流式接口外，统一使用 ApiResponse 返回结构。
"""

from fastapi import APIRouter, Request

from nexus.common.exceptions import AuthError
from nexus.api.response.response import ApiResponse
from nexus.services.auth_service import check_login
from nexus.core.skills.runtime import SkillRuntime

# 路由分组：对外路径前缀为 /fastflow/nexus/v1/slash
router = APIRouter(prefix="/slash", tags=["slash"])


@router.get("/catalog", response_model=ApiResponse)
async def get_slash_catalog(request: Request):
    """
    返回 slash 面板目录：
    - skills: 本地可用 skill 列表
    - mcp: 占位分组（后续扩展）
    """
    # 1) 从请求头读取登录态。
    auth_token = request.headers.get("Authorization")
    if not auth_token:
        raise AuthError("Authorization token is required")
    # 2) 复用现有认证服务校验登录状态。
    if not check_login(auth_token):
        raise AuthError("Current user is not logged in")

    # 3) 读取本地 skills 目录，返回可见条目。
    skill_runtime = SkillRuntime()
    skills = skill_runtime.list_skills()

    # 4) MCP 先返回占位分组：前端可以先完成交互布局，后续再接真实 MCP 目录。
    mcp_placeholder = [
        {
            "key": "mcp",
            "name": "MCP",
            "description": "MCP 选择即将开放",
            "disabled": True,
        }
    ]

    # 5) 按统一结构返回：{code, message, data}。
    return {
        "data": {
            "skills": skills,
            "mcp": mcp_placeholder,
        }
    }
